Monster jump

Verzijl opened the pull request:
The monsters/walkers are now jumping randomly around the map.

Verzijl committed:
Made a start on letting the monsters jump.

Need an improvement for the collisions as now when a monster moves up and runs into a side wall it goes through it and off the screen.

<feature, bug

Verzijl committed:
Fixed the jumping of the monster.

They now don't go out of the boundaries of the map. The jumping is now
more smooth.

<feature, bug

jimhommes commented:
Please write some tests as the coverage of Walker.java went from 75% to 66%. 

If you do that, :shipit:

toinehartman commented on src/test/java/model/WalkerTest.java:
```diff
@@ -54,10 +54,10 @@ public void testSwitchingDirectionTrue() {
 	public void testMove() throws Exception {
 		walker.move();
-		assertEquals(Settings.MONSTER_SPEED , walker.getX(), 0);
+		assertEquals(37.0 , walker.getX(), 0);
```
Why would the x coordinate be different now? Jumping is only vertical and looking at the code, it does not influence speed in the x direction. This probably means you broke something, adjusting the test is not always the right solution if it fails.

toinehartman commented on src/main/java/model/Walker.java:
```diff
@@ -54,19 +110,55 @@ public void move() {
+            setY(walkerMaxY - getHeight());
+        }
+    }
+
```
This looks like the code is inspired by the jumping of the player. This makes a lot of duplicate code, which should be put in GravityObject.

toinehartman commented on src/main/java/model/Walker.java:
```diff
@@ -96,4 +188,14 @@ private void moveHorizontal() {
+        int min = 1;
+        int max = 200;
+        int res = rand.nextInt((max - min) + 1) + min;
+        System.out.println(res);
```
Please take out prints used for debugging. If you really need to display this information, use the Logger and a String that makes clear what this number means.

toinehartman commented on src/main/java/model/Walker.java:
```diff
@@ -46,6 +91,17 @@ public Walker(double x,
+                jumpCounter++;
+            }
+
+            if (jumpCounter == 12) {
```
This condition can never be true if the first one is, so use an `else if`.

toinehartman committed:
Merge remote-tracking branch 'origin/development' into monster_jump

Verzijl committed:
Refactored to remove duplicate code.

Repositioned code so it is not duplicate.

<organisation, naming

Verzijl committed:
Added some explanation to walker test.

<documentation

Verzijl committed:
Walker jumping independent of speed.

The walkers now jump the same height and it does not depent on it's
speed.

<bug

Verzijl committed:
Fixed the tests.

<bug, testing

JPluim commented:
When monsters are walking in the top of the screen, it looks like they can jump pretty high. It doesn't hurt, but will add as an issue for further improvements. :shipit:
